glyphs2fontir/src/source.rs

```diff
@@ -2052,4 +2052,11 @@ mod tests {
         // this is a spacing-combining mark, so we shouldn't zero it's width
         assert!(glyph.default_instance().width != 0.0);
     }
+
+    // this font causes a crash when we try to generate the notdef glyph, due to
+    // funny phantom locations?
```

Comment:
Chipping in in case this is still causing bother, as I hit something similar with the vertical metrics: this can happen when the font has sparse layers, as there will be outline variations at locations where metrics are not defined in the sources.

[ufo2ft works around this by leaving .notdef undefined at sparse layers](https://github.com/googlefonts/ufo2ft/blob/b3895a96ca910c1764df016bfee4719448cfec4a/Lib/ufo2ft/util.py#L478-L507), but as we are are variable-first here we can go one better and only add .notdef outlines at locations where `Ascender` and `Descender` have entries.

(The intersection of the `Ascender` and `Descender` locations is probably enough to pass everywhere on crater, but the ideal implementation would use the union and interpolate if necessary to fill the gaps)